Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -3312,9 +3320,26 @@ else if (thisReq_ instanceof ServerRequestLogout) {
                                     initState_ = SESSION_STATE.INITIALISED;
                                     // Publish success to listeners
                                     thisReq_.onRequestSucceeded(serverResponse, branchReferral_);
-
                                     isInitReportedThroughCallBack = ((ServerRequestInitSession) thisReq_).hasCallBack();
-                                    checkForAutoDeepLinkConfiguration();
+
+                                    boolean isActivityEnabledForPromoView = true;
+                                    boolean isPromoViewShowing = false;
+                                    PromoViewHandler promoViewHandler = PromoViewHandler.getInstance();
+                                    promoViewHandler.saveAppPromoViews();
+                                    if(currentActivityReference_.get() instanceof IPromoViewControl){
+                                        isActivityEnabledForPromoView = !((IPromoViewControl)currentActivityReference_.get()).skipPromoViewsOnThisActivity();
+                                    }
+                                    if(isActivityEnabledForPromoView) {
+                                        isPromoViewShowing = promoViewHandler.showPromoView(((ServerRequestInitSession) thisReq_).getPromoActonName(),
```

Comment:
As per this implementation  it shows only one promo view dialog at a time. If tried to create another   dialog when  a promo view is already in view error will be returned  on callback. Do you think we may have a case of queue up promo views?